pupil_src/shared_modules/video_capture/file_backend.py

```diff
@@ -307,64 +318,59 @@ def idx_to_pts(self, idx):
 
     @ensure_initialisation()
     def get_frame(self):
-        while 1:
+        try:
             target_entry = self.videoset.lookup[self.target_frame_idx]
-            if target_entry.container_idx == -1:
-                self.current_frame_idx = self.target_frame_idx
-                self.target_frame_idx += 1
-                return FakeFrame(target_entry.timestamp, self.target_frame_idx)
-            elif target_entry.container_idx != self.current_container_index:
-                self.setup_video(target_entry.container_idx)
-            try:
-                av_frame = next(self.next_frame)
-                index = self._convert_frame_index(av_frame.pts)
-                if index == self.target_frame_idx:
-                    break
-                elif index < self.target_frame_idx:
-                    pass
-            except StopIteration as stop:
-                raise EndofVideoError from stop
-
-            # if self.loop:
-            #     logger.info("Looping enabled. Seeking to beginning.")
-            #     self.seek_to_frame(0)
-            #     return self.get_frame()
-            # else:
-            #     logger.debug(
-            #         "End of videofile %s %s"
-            #         % (self.current_frame_idx, len(self.timestamps))
-            #     )
-            #     raise EndofVideoError("Reached end of video file")
-
+        except IndexError:
+            raise EndofVideoError
+        if target_entry.container_idx == -1:
+            return self._get_fake_frame_and_advance(target_entry.timestamp)
+        elif target_entry.container_idx != self.current_container_index:
+            self.setup_video(target_entry.container_idx)
+        try:
+            av_frame = next(self.next_frame)
+            if not av_frame:
+                raise EndofVideoError
+            index = self._convert_frame_index(av_frame.pts)
+        except StopIteration as stop:
+            return self._get_fake_frame_and_advance(target_entry.timestamp)
+            raise EndofVideoError from stop
```

Comment:
Do we still need this line? And should be returning a fake frame? What if `fill_gaps==False`?

Reply:
Did you mean 
    `return self._get_fake_frame_and_advance(target_entry.timestamp)`
Yes, I think we don't need it anymore.

Reply:
Correct, that is the line that I meant.